nf_core/download.py

```diff
@@ -779,46 +779,19 @@ def compress_download(self):
 
         # .zip files
         if self.compress_type == "zip":
-            with ZipFile(self.output_filename, "w") as zipObj:
+            with ZipFile(self.output_filename, "w") as zip_file:
                 # Iterate over all the files in directory
-                for folderName, subfolders, filenames in os.walk(self.outdir):
+                for folder_name, _, filenames in os.walk(self.outdir):
                     for filename in filenames:
                         # create complete filepath of file in directory
-                        filePath = os.path.join(folderName, filename)
+                        file_path = os.path.join(folder_name, filename)
                         # Add file to zip
-                        zipObj.write(filePath)
+                        zip_file.write(file_path)
             log.info(f"Command to extract files: [bright_magenta]unzip {self.output_filename}[/]")
 
         # Delete original files
         log.debug(f"Deleting uncompressed files: '{self.outdir}'")
         shutil.rmtree(self.outdir)
 
         # Caclualte md5sum for output file
-        self.validate_md5(self.output_filename)
-
-    def validate_md5(self, fname, expected=None):
-        """Calculates the md5sum for a file on the disk and validate with expected.
-
-        Args:
-            fname (str): Path to a local file.
-            expected (str): The expected md5sum.
-
-        Raises:
-            IOError, if the md5sum does not match the remote sum.
-        """
-        log.debug(f"Validating image hash: {fname}")
-
-        # Calculate the md5 for the file on disk
-        hash_md5 = hashlib.md5()
-        with open(fname, "rb") as f:
-            for chunk in iter(lambda: f.read(4096), b""):
-                hash_md5.update(chunk)
-        file_hash = hash_md5.hexdigest()
-
-        if expected is None:
-            log.info(f"MD5 checksum for '{fname}': [blue]{file_hash}[/]")
-        else:
-            if file_hash == expected:
-                log.debug(f"md5 sum of image matches expected: {expected}")
-            else:
-                raise IOError(f"{fname} md5 does not match remote: {expected} - {file_hash}")
+        log.info(f"MD5 checksum for '{self.output_filename}': [blue]{nf_core.utils.file_md5(self.output_filename)}[/]")
```

Comment:
I think we want to use validate_file_md5() here to validate the md5. And thus return the md5 value instead of only True.

Reply:
`file_md5("file")` does return the md5 of the "file".  Here, validate_md5 was called without the `expected` keyword and thus no validation was carried out in the code.

The logging was done before inside the validate_md5 function, but I refactored it out of the md5 validation because it was only used here and I think the logging might not be desired in other use cases of md5 validation.

Reply:
I missed that the function was called without an `expected` value, looks good then :)